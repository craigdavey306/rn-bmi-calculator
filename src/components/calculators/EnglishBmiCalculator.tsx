import { Alert, StyleSheet, Text, TextInput, View } from 'react-native';

import HeightInput from '../ui/HeightInput';
import WeightInput from '../ui/WeightInput';

import { EnglishCalculator } from '../ui/BmiCalculator';
import { Colors, Font } from '../../library';

type EnglishBmiCalculatorProps = {
  handleHeightFeetInput: (heightFeet: string) => void;
  handleHeightInchesInput: (heightInches: string) => void;
  handleWeightPoundsInput: (weightPounds: string) => void;
} & Pick<EnglishCalculator, 'heightFeet' | 'heightInches' | 'weightPounds'>;

const EnglishBmiCalculator: React.FC<EnglishBmiCalculatorProps> = ({
  handleHeightFeetInput,
  handleHeightInchesInput,
  handleWeightPoundsInput,
  heightFeet,
  heightInches,
  weightPounds,
}) => {
  return (
    <View>
      <HeightInput
        handleFeetInput={handleHeightFeetInput}
        handleInchesInput={handleHeightInchesInput}
        heightInFeet={heightFeet.toString()}
        heightInInches={heightInches.toString()}
      />
      <WeightInput
        handleWeightInput={handleWeightPoundsInput}
        weightInPounds={weightPounds.toString()}
      />
    </View>
  );
};

export default EnglishBmiCalculator;

const styles = StyleSheet.create({
  label: {
    color: Colors.Green.grade70,
    fontSize: Font.size.size7,
    fontWeight: 'bold',
    margin: 5,
    textAlign: 'center',
  },
});
