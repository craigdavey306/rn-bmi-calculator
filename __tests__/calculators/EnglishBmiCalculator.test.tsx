import 'react-native';
import renderer from 'react-test-renderer';
import { expect, it } from '@jest/globals';
import EnglishBmiCalculator from '../../src/components/calculators/EnglishBmiCalculator';

it('Renders the English BMI Calculator correctly', () => {
  const stub = (input: string) => {};
  const calculator = (
    <EnglishBmiCalculator
      heightFeet=""
      heightInches=""
      weightPounds=""
      handleHeightFeetInput={stub}
      handleHeightInchesInput={stub}
      handleWeightPoundsInput={stub}
    />
  );

  const tree = renderer.create(calculator).toJSON();

  expect(tree).toMatchSnapshot();
});
