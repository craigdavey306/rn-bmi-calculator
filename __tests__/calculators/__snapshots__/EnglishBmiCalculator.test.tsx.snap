// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the English BMI Calculator correctly 1`] = `
<View>
  <View
    style={
      {
        "padding": 10,
      }
    }
  >
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#234e27",
          "fontSize": 18,
          "fontWeight": "bold",
          "margin": 5,
          "textAlign": "center",
        }
      }
    >
      Height
    </Text>
    <View
      style={
        {
          "marginVertical": 8,
        }
      }
    >
      <Text
        accessibilityRole="text"
        nativeID="feet"
        style={
          {
            "color": "#000000",
            "fontSize": 13,
            "marginBottom": 4,
            "marginLeft": 3,
          }
        }
      >
        Feet
      </Text>
      <TextInput
        accessibilityLabelledBy="feet"
        accessibilityRole="none"
        accessible={true}
        autoCapitalize="none"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        style={
          [
            {
              "backgroundColor": "#dfeedf",
              "borderColor": "#000000",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 13,
              "paddingHorizontal": 6,
              "paddingVertical": 8,
            },
          ]
        }
        testID="feet:testid"
        value=""
      />
    </View>
    <View
      style={
        {
          "marginVertical": 8,
        }
      }
    >
      <Text
        accessibilityRole="text"
        nativeID="inches"
        style={
          {
            "color": "#000000",
            "fontSize": 13,
            "marginBottom": 4,
            "marginLeft": 3,
          }
        }
      >
        Inches
      </Text>
      <TextInput
        accessibilityLabelledBy="inches"
        accessibilityRole="none"
        accessible={true}
        autoCapitalize="none"
        keyboardType="number-pad"
        maxLength={2}
        onChangeText={[Function]}
        style={
          [
            {
              "backgroundColor": "#dfeedf",
              "borderColor": "#000000",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 13,
              "paddingHorizontal": 6,
              "paddingVertical": 8,
            },
          ]
        }
        testID="inches:testid"
        value=""
      />
    </View>
  </View>
  <View
    style={
      {
        "padding": 10,
      }
    }
  >
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#234e27",
          "fontSize": 18,
          "fontWeight": "bold",
          "margin": 5,
          "textAlign": "center",
        }
      }
    >
      Weight
    </Text>
    <View
      style={
        {
          "marginVertical": 8,
        }
      }
    >
      <Text
        accessibilityRole="text"
        nativeID="pounds"
        style={
          {
            "color": "#000000",
            "fontSize": 13,
            "marginBottom": 4,
            "marginLeft": 3,
          }
        }
      >
        Pounds
      </Text>
      <TextInput
        accessibilityLabelledBy="pounds"
        accessibilityRole="none"
        accessible={true}
        autoCapitalize="none"
        keyboardType="numeric"
        maxLength={8}
        onChangeText={[Function]}
        style={
          [
            {
              "backgroundColor": "#dfeedf",
              "borderColor": "#000000",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 13,
              "paddingHorizontal": 6,
              "paddingVertical": 8,
            },
          ]
        }
        testID="pounds:testid"
        value=""
      />
    </View>
  </View>
</View>
`;
